src/Build/Evaluation/Evaluator.cs

```diff
@@ -2370,12 +2370,15 @@ private List<ProjectRootElement> ExpandAndLoadImports(string directoryOfImportin
 
             if (importElement.ParsedSdkReference != null)
             {
-                // Try to get the solution path when available.
+                // Try to get the path to the solution and project being built. If the solution path equals "*Undefined*" 
+                // (hard coded in Microsoft.Common.CurrentVersion.targets) set to null for consistency.
                 var solutionPath = _data.GetProperty(SolutionProjectGenerator.SolutionPathPropertyName)?.EvaluatedValue;
+                var projectPath = _data.GetProperty(ReservedPropertyNames.projectFullPath)?.EvaluatedValue;
```

Comment:
Is getting this out of the reserved property the only way to get the information at this layer? I don't like that (even though AFAIK it's immutable it seems unclean).